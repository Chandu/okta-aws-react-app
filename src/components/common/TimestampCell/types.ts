import { Table } from '@medly-components/core';
import { GetComponentProps } from '@medly-components/utils';

export type ComponentType = GetComponentProps<typeof Table>['columns'][0]['component'];
