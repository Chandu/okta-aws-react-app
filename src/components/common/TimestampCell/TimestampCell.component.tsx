import { TableColumnConfig, Text } from '@medly-components/core';
import { formatAsTimestamp } from '@utils';
import React from 'react';

export const TimestampCell: Required<TableColumnConfig>['component'] = React.memo(({ data }) => <Text>{formatAsTimestamp(data)}</Text>);
TimestampCell.displayName = 'TimestampCell';
