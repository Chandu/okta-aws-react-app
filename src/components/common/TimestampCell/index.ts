export { TimestampCell as default } from './TimestampCell.component';
