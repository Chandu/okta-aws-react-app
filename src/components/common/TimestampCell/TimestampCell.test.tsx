import { render } from '@test-utils';
import React from 'react';
import { TimestampCell } from './TimestampCell.component';

describe('TimestampCell', () => {
    it('should render same day as UTC when time exceeds time difference', () => {
        const { container } = render(<TimestampCell data={'2019-12-13T09:14:16.558+0000'} />);
        expect(container).toMatchSnapshot();
    });

    it('should render previous day of UTC when time has not exceeded time difference', () => {
        const { container } = render(<TimestampCell data={'2019-12-13T01:14:16.558+0000'} />);
        expect(container).toMatchSnapshot();
    });

    it('should render same day when timezone is already in NY', () => {
        const { container } = render(<TimestampCell data={'2019-12-13T01:14:16.558-05:00'} />);
        expect(container).toMatchSnapshot();
    });

    it('should render empty string when date is invalid', () => {
        const { container } = render(<TimestampCell data={''} />);
        expect(container).toMatchSnapshot();
    });
});
