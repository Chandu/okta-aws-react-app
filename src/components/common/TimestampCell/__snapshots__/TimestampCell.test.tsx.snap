// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimestampCell should render empty string when date is invalid 1`] = `
.c0 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  text-align: initial;
  display: inline-block;
}

<div>
  <span
    class="c0"
  />
</div>
`;

exports[`TimestampCell should render previous day of UTC when time has not exceeded time difference 1`] = `
.c0 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  text-align: initial;
  display: inline-block;
}

<div>
  <span
    class="c0"
  >
    12/12/2019 08:14:16
  </span>
</div>
`;

exports[`TimestampCell should render same day as UTC when time exceeds time difference 1`] = `
.c0 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  text-align: initial;
  display: inline-block;
}

<div>
  <span
    class="c0"
  >
    12/13/2019 04:14:16
  </span>
</div>
`;

exports[`TimestampCell should render same day when timezone is already in NY 1`] = `
.c0 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  text-align: initial;
  display: inline-block;
}

<div>
  <span
    class="c0"
  >
    12/13/2019 01:14:16
  </span>
</div>
`;
